### PR TITLE
(DOCSP-25159) [BAAS] Access App metadata with function context

### PR DESCRIPTION
**Pull request by nlarew**
## Pull Request Info

- Adds `context.app` to Function Context
- Do not merge until https://jira.mongodb.org/browse/BAAS-17460 is live

### Jira

- (DOCSP-25159) [BAAS] Access App metadata with function context

### Staged Changes (Requires MongoDB Corp SSO)

- [Functions > Context](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/context-app/functions/context/#std-label-context-app)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-app-services/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/functions/context.txt [hunk 1]
```diff
@@ -52,6 +55,102 @@ The ``context`` object exposes the following interfaces:
...
+       "providerRegion": string,
+       "model": string,
+     },
+     "lastDeployed": string,
+     "hostingUri": string,
+   }
+
+.. method:: context.app.id
+   
+   The ID of the Atlas App the contains the function.
```
Comment (krollins-mdb) [inline source/functions/context.txt:83]: [typo] "the" should probably be "that" and update to just "App"
```suggestion
   The ID of the App that contains the function.
```

File: source/functions/context.txt [hunk 2]
```diff
@@ -52,6 +55,102 @@ The ``context`` object exposes the following interfaces:
...
+
+   .. example::
+      
+      .. code-block:: json
+         
+         "60c8e59866b0c33d14ee634a"
+
+.. method:: context.app.projectId
+
+   The ID of the Atlas Project/Group that contains the App.
```
Comment (krollins-mdb) [inline source/functions/context.txt:93]:
```suggestion
   The :ref:`ID of the Atlas Project/Group <find-your-project-or-app-id>` that contains the App.
```

File: source/functions/context.txt [hunk 3]
```diff
@@ -52,6 +55,102 @@ The ``context`` object exposes the following interfaces:
    * - :ref:`context.values <context-values>`
      - Contains static global :ref:`values <values-and-secrets>`.
 
+.. _context-app:
+
+Get App Metadata (``context.app``)
+----------------------------------
+
+The ``context.app`` object contains metadata about the Atlas App that
```
Comment (krollins-mdb) [inline source/functions/context.txt:63]:
```suggestion
The ``context.app`` object contains metadata about the App that
```

File: source/functions/context.txt [hunk 4]
```diff
@@ -52,6 +55,102 @@ The ``context`` object exposes the following interfaces:
...
+
+   .. example::
+      
+      .. code-block:: json
+         
+         "5e1ec444970199272441a214"
+
+.. method:: context.app.name
+
+   The name of the Atlas App that contains the function.
```
Comment (krollins-mdb) [inline source/functions/context.txt:103]:
```suggestion
   The name of the App that contains the function.
```

File: source/functions/context.txt [hunk 5]
```diff
@@ -52,6 +55,102 @@ The ``context`` object exposes the following interfaces:
...
+
+   .. example::
+      
+      .. code-block:: json
+         
+         "myapp"
+
+.. method:: context.app.clientAppId
+   
+   The unique App ID for the Atlas App that contains the function.
```
Comment (krollins-mdb) [inline source/functions/context.txt:113]:
```suggestion
   The unique App ID for the App that contains the function.
```

File: source/functions/context.txt [hunk 6]
```diff
@@ -52,6 +55,102 @@ The ``context`` object exposes the following interfaces:
...
+      .. code-block:: json
+         
+         {
+           "providerRegion": "aws-us-east-1",
+           "model": "GLOBAL"
+         }
+
+.. method:: context.app.lastDeployed
+   
+   The date and time that the Atlas App was last deployed formatted as an ISODate string.
```
Comment (krollins-mdb) [inline source/functions/context.txt:136]:
```suggestion
   The date and time that the App was last deployed, formatted as an ISODate string.
```

File: source/functions/context.txt [hunk 7]
```diff
@@ -52,6 +55,102 @@ The ``context`` object exposes the following interfaces:
...
+   
+   .. example::
+      
+      .. code-block:: javascript
+         
+         "2022-10-31T12:00:00.000Z"
+
+.. method:: context.app.hostingUri
+   
+   The base URL of :ref:`static hosting <static-hosting>` assets, if enabled.
```
Comment (krollins-mdb) [inline source/functions/context.txt:146]: Style guide says we should put conditionals at the beginning of a sentence so readers can skip it if it doesn't apply.
```suggestion
   If :ref:`static hosting <static-hosting>` is enabled, this value is the base URL for assets.
```

**Review by krollins-mdb — COMMENTED** [review]
Hi Nick! I noticed a few things in this PR that would be good to address before merging.

- It would be good to order the object properties alphabetically and also their definition sections. Then we maintain a parallel structure between the two and readers can anticipate where to find content.
- There are a lot of instances of lowercased "function". These should be capitalized if they're referring to an Atlas Function.
- On line 20, we currently have "... in your app". This should be updated to "... in your App Services App". Then, later in the page we can refer to it as just "App".

**Comment by nlarew** [comment]
@krollins-mdb Updated per your comments - thanks!

> It would be good to order the object properties alphabetically and also their definition sections

I'm a bit torn on this. I see the value of alphabetical order but as-is I've organized with a bit more of an editorial touch. i.e. I think these are roughly grouped by relation to eachother and ordered by importance to the reader.

For example, here's the alphabetized object:
```typescript
{
  "clientAppId": string,
  "deployment": {
    "model": string,
    "providerRegion": string,
  },
  "hostingUri": string,
  "id": string,
  "lastDeployed": string,
  "name": string,
  "projectId": string,
}
```

Putting the App's name and ID (likely the most commonly needed values) after `hostingUri` (a value for a feature we'll likely deprecate and that most apps don't use) feels off to me. Thoughts?

**Comment by krollins-mdb** [comment]
@nlarew, that makes sense! If you have a good understanding of what's most important to readers, then we should absolutely put those first. I think as long as the following sections follow the same order, we're all set! That'll help readers know where to look on the page for the property they're interested in.

**Review by krollins-mdb — APPROVED** [review]
Nice! This looks great. 🚢 it!

**Pull request merged**